### PR TITLE
PM-1764 - Add daily Seed Synchronisation Check

### PR DESCRIPTION
**Pull request by kaozenn**

### CODE REVIEW

File: .github/workflows/sync-check.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    branches: [PM-1764-create-daily-seed-synchronisation-check]
+  workflow_dispatch:
+  schedule:
+      - cron: '0 0 * * 2,5' # At 00:00 on Tuesday, and Friday
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
+  cancel-in-progress: true
+jobs:
+  parse-mainnet-seed-list:
+    runs-on: minafoundation-xlarge-runners
```
Comment (simisimis) [inline .github/workflows/sync-check.yaml:13]: question: any reason you chose to use xlarge runner just to do `cat networks/mainnet.txt`? seems a bit excessive

**Review by simisimis — APPROVED** [review]
other than non blocking question, looks good!

File: .github/workflows/sync-check.yaml (hunk 1 above)
Comment (kaozenn) [reply .github/workflows/sync-check.yaml:13]: It's a good catch @simisimis. I originally put this runner as we had issue with the `minafoundation-default-runners`. I am reverting back!

**Review by Smorci — APPROVED** [review]

File: networks/mainnet.txt
Comment (johnmarcou) [inline networks/mainnet.txt:1]: **question:** Why do we have changes to the seed list? I think we should rebase the PR onto `main`.

File: scripts/mina-sync-monitor.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  check_sync_status
+  if [ "$status" == "Synced" ]; then
+    echo "Mina client is synced."
+    exit_code=0
+    break
+  fi
+
+  attempt=$((attempt + 1))
+
+  echo "Mina client is not synced. Attempt $attempt/$max_attempts."
```
Comment (johnmarcou) [inline scripts/mina-sync-monitor.sh:24]: **nitpick:**
```suggestion
  echo "Mina daemon is not synced. Attempt $attempt/$max_attempts."
```

File: scripts/mina-sync-monitor.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+set -e
+exit_code=1
```
Comment (johnmarcou) [inline scripts/mina-sync-monitor.sh:3]: **nitpick:** I suggest to remove this `exit_code` var, and use directly `exit 0` or `exit 1` where required.

```suggestion
```

File: scripts/mina-sync-monitor.sh [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+check_sync_status() {
+  status=$(docker exec mina mina client status --json | jq -r .sync_status || echo "Null")
+  echo "Current sync status: $status"
+}
+max_attempts=24
+attempt=0
+sleep_duration=300
```
Comment (johnmarcou) [inline scripts/mina-sync-monitor.sh:11]: **nitpick:** We could move the variables initialization to the top, and also add `status="Null"`.

**Review by johnmarcou — APPROVED** [review]
LGTM!

File: scripts/mina-sync-monitor.sh (hunk 3 above)
Comment (kaozenn) [reply scripts/mina-sync-monitor.sh]: Code updated

File: .github/workflows/sync-check.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
+name: Daily Seed Synchronization Check
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * 2,5" # At 00:00 on Tuesday, and Friday
```
[the remark this answers is not in the document]
Comment (kaozenn) [reply .github/workflows/sync-check.yaml]: Good Idea

**Pull request merged**